cli/pkg/release/http_test.go

```diff
@@ -0,0 +1,209 @@
+				"application":            {"potato"},
+			},
+			expectedMultupartFormFile: map[string][]simpleMultipartFormFileHeader{
+				"manifests[development]": {
+					{
+						filename: "development-manifest",
+						content: "some development manifest",
+					},
+				},
+			},
+
+			responseCode: http.StatusOK,
+		},
+		{
+			name: "multiple environment manifests",
+			params: &ReleaseParameters{
+				Application: "potato",
+				Manifests: map[string]string{
+					"development": "some development manifest",
+					"production":  "some production manifest",
+				},
+			},
+			expectedMultipartFormValue: map[string][]string{
+				"application": {"potato"},
+			},
+			expectedMultupartFormFile: map[string][]simpleMultipartFormFileHeader{
+				"manifests[development]": {
+					{
+						filename: "development-manifest",
+						content: "some development manifest",
+					},
+				},
+				"manifests[production]": {
+					{
+						filename: "production-manifest",
+						content: "some production manifest",
+					},
+				},
+			},
+			responseCode: http.StatusOK,
+		},
+		{
+			name: "multiple environment manifests with response code BadRequest",
+			params: &ReleaseParameters{
+				Application: "potato",
+				Manifests: map[string]string{
+					"development": "some development manifest",
+					"production":  "some production manifest",
+				},
+			},
+			expectedMultipartFormValue: map[string][]string{
+				"application": {"potato"},
+			},
+			expectedMultupartFormFile: map[string][]simpleMultipartFormFileHeader{
+				"manifests[development]": {
+					{
+						filename: "development-manifest",
+						content: "some development manifest",
+					},
+				},
+				"manifests[production]": {
+					{
+						filename: "production-manifest",
+						content: "some production manifest",
+					},
+				},
+			},
+			expectedErrorMsg: "error while issuing HTTP request, error: response was not OK or Accepted, response code: 400",
+			responseCode:     http.StatusBadRequest,
+		},
+	}
+
+	for _, tc := range tcs {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			mockServer := &mockHttpServer{
+				response: tc.responseCode,
+			}
+			server := httptest.NewServer(mockServer)
+
+			// check errors
+			err := Release(server.URL, tc.params)
+			if err != nil && err.Error() != tc.expectedErrorMsg {
+				t.Fatalf("error messages mismatched, expected %s, received %s", tc.expectedErrorMsg, err.Error())
+			}
+			if err == nil && tc.expectedErrorMsg != "" {
+				t.Fatalf("expected error %v, but no error was raised", tc.expectedErrorMsg)
+			}
+
+			// check multipart form values
+			if !cmp.Equal(mockServer.multipartForm.Value, tc.expectedMultipartFormValue) {
+				t.Fatalf("request multipart forms are different, expected %v, received %v", tc.expectedMultipartFormValue, mockServer.multipartForm)
+			}
+
+			// check multipart form files
+			fileHeaders := make(map[string][]simpleMultipartFormFileHeader)
+			for key, val := range mockServer.multipartForm.File {
```

Comment:
```suggestion
                        // we do not need to sort the map before iterating over it, because we just use `cmp.Equal` which handles the undefined order
			for key, val := range mockServer.multipartForm.File {
```